deepchecks/nlp/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -46,12 +49,15 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         """Run check."""
         dataset = context.get_data_by_kind(dataset_kind)
         model = context.model
-        scorers = context.get_scorers(self.scorers, use_avg_defaults=False)
+        span_aligner = SpanAligner()
```

Comment:
I think this should be created and employing in a more general manner, is this used only for performance checks?
Consider placing it in context (usecase in mind in for evaluation suite)

Reply:
This can be done only assuming all check run on the exact same sample. It seems reasonable so I'm willing to do it, but then we need to remember not to enable different samples for different checks

Reply:
not necessarily, you can re-initialize it if needed